packages/transformer/src/IModelTransformer.ts

```diff
@@ -1623,115 +1583,95 @@ export class IModelTransformer extends IModelExportHandler {
-   * create a [[PartiallyCommittedEntity]] to track resolution of those references
-   */
-  private collectUnmappedReferences(entity: ConcreteEntity) {
-    const missingReferences = new EntityReferenceSet();
-    let thisPartialElem: PartiallyCommittedEntity | undefined;
+  private doAllReferencesExistInTarget(entity: ConcreteEntity) {
+    let allReferencesExist = true;
+    for (const referenceId of entity.getReferenceIds()) {
+      const referencedEntityId = EntityReferences.toId64(referenceId);
+      if (
+        referencedEntityId === IModel.repositoryModelId ||
+        referencedEntityId === IModel.dictionaryId ||
+        referencedEntityId === "0xe"
+      ) {
+        continue;
+      }
 
-    // eslint-disable-next-line deprecation/deprecation
-    for (const referenceId of entity.getReferenceConcreteIds()) {
-      // TODO: probably need to rename from 'id' to 'ref' so these names aren't so ambiguous
-      const referenceIdInTarget = this.context.findTargetEntityId(referenceId);
-      const alreadyProcessed =
-        EntityReferences.isValid(referenceIdInTarget) ||
-        this._skippedEntities.has(referenceId);
-      if (alreadyProcessed) continue;
-      Logger.logTrace(
-        loggerCategory,
-        `Deferring resolution of reference '${referenceId}' of element '${entity.id}'`
-      );
-      const referencedExistsInSource = EntityUnifier.exists(this.sourceDb, {
-        entityReference: referenceId,
-      });
-      if (!referencedExistsInSource) {
-        Logger.logWarning(
-          loggerCategory,
-          `Source ${EntityUnifier.getReadableType(entity)} (${
-            entity.id
-          }) has a dangling reference to (${referenceId})`
-        );
-        switch (this._options.danglingReferencesBehavior) {
-          case "ignore":
-            continue;
-          case "reject":
-            throw new IModelError(
-              IModelStatus.NotFound,
-              [
-                `Found a reference to an element "${referenceId}" that doesn't exist while looking for references of "${entity.id}".`,
-                "This must have been caused by an upstream application that changed the iModel.",
-                "You can set the IModelTransformOptions.danglingReferencesBehavior option to 'ignore' to ignore this, but this will leave the iModel",
-                "in a state where downstream consuming applications will need to handle the invalidity themselves. In some cases, writing a custom",
-                "transformer to remove the reference and fix affected elements may be suitable.",
-              ].join("\n")
-            );
+      if (
+        allReferencesExist &&
+        !EntityReferences.isValid(this.context.findTargetEntityId(referenceId))
+      ) {
+        // TODO: do we ever care if the reference did not exist in master? why not just remove the reference if it already did nothing? is that option ever used or is useful in any way?
```

Comment:
yes, I was referring to that option, but I agree that there are cases when this may might be useful. I guess I was just not liking the option because the code would be much simpler without it.